cli/cmd/export/export.go

```diff
@@ -82,6 +93,60 @@ func (e *Export) Run(ctx *clicontext.CLIContext) error {
 	return nil
 }
 
+func collectObjectsFromStack(ctx *clicontext.CLIContext, stack *riov1.Stack) ([]runtime.Object, error) {
+	var objects []runtime.Object
+	client, err := dynamic.NewForConfig(ctx.Config.RestConfig)
+	if err != nil {
+		return nil, err
+	}
+	for _, gvk := range stack.Spec.AdditionalGroupVersionKinds {
+		gvr := schema.GroupVersionResource{
+			Group:    gvk.Group,
+			Version:  gvk.Version,
+			Resource: strings.ToLower(name.GuessPluralName(gvk.Kind)),
+		}
+		resourceClient := client.Resource(gvr)
+		ul, err := resourceClient.List(metav1.ListOptions{
+			LabelSelector: "rio.cattle.io/stack=" + stack.Name,
+		})
+		for _, item := range ul.Items {
+			data, err := item.MarshalJSON()
+			if err != nil {
+				return nil, err
+			}
+			if item.GetKind() == "Service" {
+				svc := &riov1.Service{}
+				if err := json.Unmarshal(data, svc); err != nil {
+					return nil, err
+				}
+				objects = append(objects, svc)
+			} else if item.GetKind() == "ConfigMap" {
+				cm := &corev1.ConfigMap{}
+				if err := json.Unmarshal(data, cm); err != nil {
+					return nil, err
+				}
+				objects = append(objects, cm)
+			} else if item.GetKind() == "ExternalService" {
+				es := &riov1.ExternalService{}
+				if err := json.Unmarshal(data, es); err != nil {
+					return nil, err
+				}
+				objects = append(objects, es)
+			} else if item.GetKind() == "Router" {
+				r := &riov1.Router{}
+				if err := json.Unmarshal(data, r); err != nil {
+					return nil, err
+				}
+				objects = append(objects, r)
+			}
```

Comment:
@cbron Do you export any object that is not Rio types? like any k8s manifest you can define in Riofile.

Reply:
fixed